packages/node/src/login/oidc/oidcHandlers/RefreshTokenOidcHandler.ts

```diff
@@ -56,6 +58,49 @@ function validateOptions(
   );
 }
 
+async function refreshAccess(
+  refreshOptions: RefreshOptions,
+  dpop: boolean
+): Promise<TokenSet & { fetch: typeof fetch }> {
+  let authFetch: typeof fetch;
+  // eslint-disable-next-line camelcase
+  let tokens: TokenSet & { access_token: string };
+  let dpopKey: JWK.ECKey;
+  if (dpop) {
+    dpopKey = await JWK.generate("EC", "P-256");
+    tokens = await refreshOptions.tokenRefresher.refresh(
+      refreshOptions.sessionId,
+      refreshOptions.refreshToken,
+      dpopKey
+    );
+  } else {
+    tokens = await refreshOptions.tokenRefresher.refresh(
+      refreshOptions.sessionId,
+      refreshOptions.refreshToken
+    );
+  }
+  // Rotate the refresh token if applicable
+  const rotatedRefreshOptions = {
+    ...refreshOptions,
+    refreshToken: tokens.refresh_token ?? refreshOptions.refreshToken,
+  };
+  if (dpop) {
+    authFetch = await buildDpopFetch(
+      tokens.access_token,
+      // dpopkey is assigned in the previous if block
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
Isn't it just easier to do `if (dpop && dpopKey)` above?

Reply:
Unfortunately no, it still complains that `Variable 'dpopKey' is used before being assigned.`

Reply:
Ah, I see. So I see the same `if...else` here twice - couldn't the contents of this `if` just be added to the previous one? The only resulting duplication would be lines 83-86, but I think that actually still saves more lines than it costs, and I think it's less likely that that causes mistakes in the future than e.g. the `if` clauses no longer being aligned.

Reply:
Yeah true, I guss I'll switch back to that. I initially went this way to reduce (a bit) cyclic complexity, and avoid having to write two very similar tests, one checking for token rotation with DPoP, and the other for Bearer tokens, but you're right that it's liklier that an issue is introduced if the `if` statements get disaligned ^^